conans/client/generators/cmake_find_package.py

```diff
@@ -156,10 +160,6 @@ class CMakeFindPackageGenerator(GeneratorComponentsMixin, Generator):
         set(CMAKE_MODULE_PATH {{ comp.build_paths }} ${CMAKE_MODULE_PATH})
         set(CMAKE_PREFIX_PATH {{ comp.build_paths }} ${CMAKE_PREFIX_PATH})
 
-        foreach(_BUILD_MODULE_PATH {{ '${'+pkg_name+'_'+comp_name+'_BUILD_MODULES_PATHS}' }})
-            include(${_BUILD_MODULE_PATH})
-        endforeach()
-
```

Comment:
Maybe I'm wrong, but...

If components are iterated in order, then it is safe to add these build-modules here (just after the component). That way we ensure that a build-module declared at a component level is not using any target-component that is not available.

...and we include the `build_modules` declared at component level.

Reply:
The problem is that you don't have the component target declared at this point